### ISSUE TITLE
`isPathIgnored()` should consider extension

### ISSUE THREAD

**Issue by kentor** [eslint/eslint#4462]
It seems reasonable that the `CLIEngine`'s [`isPathIgnored()`](http://eslint.org/docs/developer-guide/nodejs-api#ispathignored) method should return `true` for files that don't match the extensions given to `CLIEngine`. 

```js
> eslint = require('eslint')
> cli = new eslint.CLIEngine()
> cli.options.extensions
[ '.js' ]
> cli.isPathIgnored('.git/index')
false
```

This is causing `eslint-watch` to lint files such as `.git/index` or `npm-debug.log` when files are changed. See: https://github.com/rizowski/eslint-watch/issues/52, https://github.com/rizowski/eslint-watch/blob/4b28d0d811b2c52713b52a51c44726daf51f3f09/src/watcher.js#L36-L39

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by kentor** [comment]
sure

**Comment by nzakas** [comment]
Closing per discussion on #4465
